### PR TITLE
Add Visual Studio Code Insiders build for beta stream

### PR DESCRIPTION
**Pull request by fooishbar**
Based on the existing com.visualstudio.code build, containing the
release stream of VS Code, add a separate app-ID for the Insiders
stream, which is their beta release channel.

Disclaimers have been added to the app description and warning file
opened at first startup, however this should still only be distributed
in a beta channel, and not generally.

----

This was originally #1422, but I accidentally pushed from the wrong branch, which automatically closed the PR, and now it can never be re-opened. Oops.

### CODE REVIEW

File: com.visualstudio.code.insiders.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          - mv usr/share/code-insiders vscode-insiders
+          - rm -r code-insiders.deb control.tar.gz data.tar.xz debian-binary usr
+      - type: file
+        path: code-insiders.sh
+      - type: file
+        path: flatpak-warning.txt
+      - type: file
+        path: npmrc
+      - type: file
+        path: com.visualstudio.code.insiders.appdata.xml
```
Comment (LeandroStanger) [inline com.visualstudio.code.insiders.yaml:71]:
```suggestion
        path: com.visualstudio.code.insiders.metainfo.xml
```

Comment (nedrichards) [reply com.visualstudio.code.insiders.yaml:71]: Mate, this also needs to be fixed on line 47. I appreciate that you want to help but we need to make sure that we're helping people as efficiently as possible by testing our changes if we're going to make them as one click commit suggestions like this.

Comment (barthalion) [reply com.visualstudio.code.insiders.yaml:71]: Not to mention it's mostly irrelevant as flatpak copies the file to `appdata` directory anyway.

**Comment by barthalion** [comment]
bot, build com.visualstudio.code.insiders

File: com.visualstudio.code.insiders.yaml (hunk 1 above)
Comment (fooishbar) [reply com.visualstudio.code.insiders.yaml]: Eugh, I don't know how this crept back in - must've been juggling files between the `com.visualstudio.code` forked repo and this one. Fixed now.

**Comment by barthalion** [comment]
bot, build com.visualstudio.code.insiders

**Comment by barthalion** [comment]
`com.visualstudio.code.insiders:    AppData problem: attribute-invalid : <release> timestamp is in the future`

**Comment by fooishbar** [comment]
@barthalion Fixed. And I ran a test build to make sure ...

**Comment by barthalion** [comment]
bot, build com.visualstudio.code.insiders

**Comment by fooishbar** [comment]
I _think_ this was because I had the wrong `launchable` ID in the metainfo? Is there any way I can run the complete BuildBot checks locally, rather than just bouncing off you and a GitHub bot?

**Comment by bilelmoussaoui** [comment]
You can validate  the metainfo locally with https://github.com/flathub/flathub/wiki/AppData-Guidelines#use-flathubs-appstream-util

**Comment by fooishbar** [comment]
Yes, but that validation [did pass](https://flathub.org/builds/#/builders/12/builds/2133) - it's just other things BuildBot did afterwards (like validating icons) which failed.

**Comment by barthalion** [comment]
Yeah, we definitely need to streamline "check build locally" process. It means `appstream-compose` failed to find and copy icon, I'll check locally why that happened.

**Comment by barthalion** [comment]
bot, build com.visualstudio.code.insiders

**Comment by barthalion** [comment]
I guess a solar flare? It worked now and also works locally.

**Comment by fooishbar** [comment]
If it works off the `<launchable>` entry in the metainfo, that was wrong (`.code` rather than `.code.insiders`), which I corrected after the build failed as that was my only real guess.

Thanks a lot for all your help & patience!

**Comment by barthalion** [comment]
Ah, missed that you pushed new commits. :+1:

File: flatpak-warning.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+https://www.flathub.org
+
+------------------------------------------------------------------------------------
+| Warning: You are running an unofficial Flatpak version of Visual Studio Code !!! |
+|          Visual Studio Code Insiders is also an experimental beta release, and   |
+|          not fully tested and stable! Please switch back to the release channel  |
+|          if you are experiencing issues with this program.                       |
+------------------------------------------------------------------------------------
+
+Please open issues under: https://github.com/flathub/com.visualstudio.code/issues
```
Comment (barthalion) [inline flatpak-warning.txt:11]: It should be pointing to https://github.com/flathub/com.visualstudio.code.insiders

**Comment by barthalion** [comment]
LGTM, just needs to point to correct repo and I'm happy to merge it.

File: flatpak-warning.txt (hunk 2 above)
Comment (fooishbar) [reply flatpak-warning.txt]: Oh, good catch. That was left over from when it may have been a branch in the main repo. Fixed, thanks!

**Comment by fooishbar** [comment]
Done, thankyou!

**Pull request closed without merging**